proto/prost/private/prost.bzl

```
@@ -0,0 +1,479 @@
+"""Rules for building protos in Rust with Prost and Tonic."""
+
+load("@rules_proto//proto:defs.bzl", "ProtoInfo", "proto_common")
+load("//rust:defs.bzl", "rust_common")
+
+# buildifier: disable=bzl-visibility
+load("//rust/private:providers.bzl", "DepVariantInfo")
+
+# buildifier: disable=bzl-visibility
+load("//rust/private:rustc.bzl", "rustc_compile_action")
+load(":providers.bzl", "ProstProtoInfo", "TonicProtoInfo")
+
+RUST_EDITION = "2021"
+
+TOOLCHAIN_TYPE = "@rules_rust//proto/prost:toolchain_type"
+
+def _create_proto_lang_toolchain(ctx, prost_toolchain):
+    is_tonic = prost_toolchain.tonic_runtime != None
+
+    proto_lang_toolchain = proto_common.ProtoLangToolchainInfo(
+        out_replacement_format_flag = "--prost_out=%s",
+        plugin_format_flag = prost_toolchain.prost_plugin_flag,
+        plugin = prost_toolchain.prost_plugin[DefaultInfo].files_to_run,
+        runtime = prost_toolchain.prost_runtime,
+        provided_proto_sources = depset(),
+        proto_compiler = ctx.attr._prost_process_wrapper[DefaultInfo].files_to_run,
+        protoc_opts = prost_toolchain.protoc_opts,
+        progress_message = "",
+        mnemonic = "TonicGenProto" if is_tonic else "ProstGenProto",
```

```suggestion
    mnemonic = "TonicGenProto" if is_tonic else "ProstGenProto"
    proto_lang_toolchain = proto_common.ProtoLangToolchainInfo(
        out_replacement_format_flag = "--prost_out=%s",
        plugin_format_flag = prost_toolchain.prost_plugin_flag,
        plugin = prost_toolchain.prost_plugin[DefaultInfo].files_to_run,
        runtime = prost_toolchain.prost_runtime,
        provided_proto_sources = depset(),
        proto_compiler = ctx.attr._prost_process_wrapper[DefaultInfo].files_to_run,
        protoc_opts = prost_toolchain.protoc_opts,
        progress_message = mnemonic + " %{label}",
        mnemonic = mnemonic,
```

To avoid outputting ": " as the progress message in the terminal.